### PR TITLE
Med 54 local doctor profile UI

### PR DESCRIPTION
**Pull request by eduardojigub**
Add profile doctor page

Route `/profile`
1. Add inputs and form control on Profile Page
2. Avatar from antD,
3. Button to change profile picture 

**PR checklist:**

**1.1 Common**
- [x] 1.1.2 no any, should be also added to eslint rules
- [x] 1.1.1 types for input and output params
- [x] 1.1.3 try/catch for any async function
- [x] 1.1.4 no magic numbers
- [x] 1.1.5 compare only with constants not with strings (instead of user === ‘admin’, better user === roles.admin)
- [x] 1.1.6 no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] 1.1.7 avoid similar types with duplicating by generics
- [x] 1.1.8 no console.log in pr
- [x] 1.1.9 .env file should be in .gitignore
- [x] 1.1.10 delete commented code if it's not part of documentation
- [x] 1.1.11 no links in the code, env links should be in env file (for example: server url),
- [x] constant links (for example default avatar URL) should be in constant file
- [x] 1.1.12 constants and function names should be lowercase.
- [x] 1.1.13 no dates format in the code (like "dd/MM/yyyy”), move it in global constant variable
- [x] 1.1.14 import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] 1.1.15 strings should be in the constant variable. Example: instead of ‘15 3 * * *’, const EACH_DAY_15h_15min = ‘15 3 * * *’

 **1.3 Frontend|Mobile**
- [x] 1.3.1 split component and logic (move logic to custom hook file)
- [x] 1.3.2 colors, font size, and font name should be in the theme or in the constants
- [x] 1.3.3 no text in the components, use i18n approach, even if you have only one language for now
- [x] 1.3.4 inline styles prohibited
- [x] 1.3.5 import should be absolute. instead of ../../../components/myComponent should be components/myComponent

image

### CODE REVIEW

File: apps/medon-fe/src/assets/ReactComponentAssets/ChangeProfileAvatar.tsx [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { ReactComponent as ChangeProfilePictureSvg } from '../editPhotoIcon.svg';
+
+export default ChangeProfilePictureSvg;
```
Comment (lyudad) [inline apps/medon-fe/src/assets/ReactComponentAssets/ChangeProfileAvatar.tsx:3]: what is the reason to have this file?

File: apps/medon-fe/src/components/ProfileForm/AvatarUploader/index.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import ChangeProfilePictureSvg from 'assets/ReactComponentAssets/ChangeProfileAvatar';
+import { useTheme } from 'styled-components';
+
+
+export const AvatarUploader = () => {
+  const [avatarFile, setAvatarFile] = useState<RcFile | undefined>(undefined);
+  const theme = useTheme();
+
+  const handleAvatarChange = (info: UploadChangeParam) => {
+    if (info.file.status === 'done') {
```
Comment (lyudad) [inline apps/medon-fe/src/components/ProfileForm/AvatarUploader/index.tsx]: check checklist 1.1.5

File: apps/medon-fe/src/components/ProfileForm/index.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { BrowserRouter } from 'react-router-dom';
+import '../../translation/i18next';
```
Comment (lyudad) [inline apps/medon-fe/src/components/ProfileForm/index.spec.tsx]: 1.3.5

File: apps/medon-fe/src/components/ProfileForm/styles.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  line-height: 1.5;
+  letter-spacing: 0em;
+`;
+
+export const ErrorMsg = styled.p`
+  position: absolute;
+  margin-top: 0.2rem;
+  width: 13rem;
+  font-family: ${({ theme }) => theme.typography.fontFamily.sf_pro_text};
+  font-size: 0.8rem;
```
Comment (lyudad) [inline apps/medon-fe/src/components/ProfileForm/styles.tsx]: it should be in constants 1.3.2

File: apps/medon-fe/src/components/ProfileForm/index.spec.tsx (hunk 3 above)
Comment (eduardojigub) [reply apps/medon-fe/src/components/ProfileForm/index.spec.tsx]: Solved

File: apps/medon-fe/src/assets/ReactComponentAssets/ChangeProfileAvatar.tsx (hunk 1 above)
Comment (eduardojigub) [reply apps/medon-fe/src/assets/ReactComponentAssets/ChangeProfileAvatar.tsx:3]: I'm transforming the svg into a react component and storing it inside the assets folder, but in a exclusive folder to deal only with that called "ReactComponentAssets". Is there a better way to deal with it? Thanks in advance

File: apps/medon-fe/src/components/ProfileForm/AvatarUploader/index.tsx (hunk 2 above)
Comment (eduardojigub) [reply apps/medon-fe/src/components/ProfileForm/AvatarUploader/index.tsx]: Solved

**Pull request closed without merging**